### PR TITLE
feat(mobile - dart knight): implement the homepage 

### PR DESCRIPTION
**Pull request by SimonMekonnen**
Title: Add homepage

Description:
this pull request is to add the homepage to the project. the homepage is a simple page that shows different articles.

Changes Made:
      - create the filter buttons for the homepage
      - create the profile picture as a widget
      - create a card for holding the article info and image 
      - create the search bar for the homepage
      - customize fonts
      - layout all the components 
Screenshots:
photo_2023-04-25_00-37-27

### CODE REVIEW

File: Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/articleImage_widget.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            width: 110,
+            height: 30,
+            decoration: BoxDecoration(
+                color: Colors.white, borderRadius: BorderRadius.circular(10)),
+            child: const Center(
+                child: Text("5 min read",
+                    style: TextStyle(
+                      fontFamily: "Poppins",
+                      fontWeight: FontWeight.w500,
+                      color: Color(0xFF424242),
```
Comment (lealemgetahun) [inline Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/articleImage_widget.dart:32]: Avoid using fixed colors. Have a separate file in /core/utils and define the colors there.

File: Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/articleImage_widget.dart
Comment (lealemgetahun) [inline Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/articleImage_widget.dart:1]: use small_snack_case for file name. Rename it to articel_image_widget

File: Mobile/starter_project_dark_knights/assets/images/do.jpg
Comment (lealemgetahun) [inline Mobile/starter_project_dark_knights/assets/images/do.jpg:1]: use descriptive names for the images

**Review by lealemgetahun — CHANGES_REQUESTED** [review]

File: Mobile/starter_project_dark_knights/lib/features/feeds/presentation/screen/home_page.dart [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                    shape: RoundedRectangleBorder(
+                      borderRadius: BorderRadius.circular(12),
+                    ),
+                  ),
+                  onPressed: () {},
+                  child: const Icon(Icons.add)),
+            ),
+          )
+
+          // const ArticleCard()
```
[the remark this answers is not in the document]
Comment (SimonMekonnen) [reply Mobile/starter_project_dark_knights/lib/features/feeds/presentation/screen/home_page.dart]: Done!

File: Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/filters_widget.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    final screenWidth = MediaQuery.of(context).size.width;
+    return FilterChip(
+      backgroundColor: active == false ? Colors.white : const Color(0xFF376AED),
+      labelPadding:
+          EdgeInsets.fromLTRB(screenWidth * 0.03, 0, screenWidth * 0.03, 0),
+      label: Text(lable,
+          style: TextStyle(
+              fontFamily: "Poppins",
+              fontWeight: FontWeight.w500,
+              color: active == false ? const Color(0xFF376AED) : Colors.white)),
```
[the remark this answers is not in the document]
Comment (SimonMekonnen) [reply Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/filters_widget.dart]: Done!

File: Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/articlecard_widget.dart
[the remark this answers is not in the document]
Comment (SimonMekonnen) [reply Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/articlecard_widget.dart:1]: Done!

File: Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/articleInfo_widget.dart
[the remark this answers is not in the document]
Comment (SimonMekonnen) [reply Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/articleInfo_widget.dart:1]: Done!

File: Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/articleImage_widget.dart
Comment (SimonMekonnen) [reply Mobile/starter_project_dark_knights/lib/features/feeds/presentation/widgets/articleImage_widget.dart:1]: Done!

File: Mobile/starter_project_dark_knights/assets/images/do.jpg
Comment (SimonMekonnen) [reply Mobile/starter_project_dark_knights/assets/images/do.jpg:1]: Done!

File: Mobile/starter_project_dark_knights/lib/features/feeds/presentation/screen/home_page.dart [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                ],
+              ),
+            ),
+            const SearchBar(),
+            Padding(
+              padding: EdgeInsets.all(screenWidth * 0.02),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                children: [
+                  Filters(
```
Comment (Redi22) [inline Mobile/starter_project_dark_knights/lib/features/feeds/presentation/screen/home_page.dart:51]: let us use builders for this.

**Review by Redi22 — CHANGES_REQUESTED** [review]

**Review by Redi22 — APPROVED** [review]
LGTM

**Review by lealemgetahun — APPROVED** [review]

**Review by Redi22 — APPROVED** [review]
LGTM

**Pull request merged**